### PR TITLE
Add cache system documentation

### PR DESCRIPTION
**Pull request by octonawish-akcodes**
Added cache system documentation, Need reviews @spbnick

### CODE REVIEW

File: doc/administrator_guide.md [hunk 1]
```diff
@@ -240,4 +292,3 @@ To upgrade the dataset schema, do the following.
         kcidb-db-dump -d bigquery:kernelci01_archive > kernelci01_archive.json
         # Using new-schema kcidb
         kcidb-db-load -d bigquery:kernelci02 < kernelci01_archive.json
-
```
Comment (spbnick) [inline doc/administrator_guide.md:243]: Could you not remove this line? That would be unrelated to your actual change.

File: doc/administrator_guide.md [hunk 2]
```diff
@@ -52,6 +56,54 @@ notifications from the Firestore collection, and sends them out through GMail,
...
+
+1. **User File Request**: When a user requests a file then that request is directed to the `kcidb_cache_redirect()` cloud function. This function serves as the entry point for processing user requests and initiating the cache lookup process.
+
+2. **Cache Lookup**: The `kcidb_cache_redirect()` function interacts with the Google Cloud Storage Bucket to find and serve the file from there, if its available.
+
+3. **File Availability Check**: If the requested file is found within the cache storage, the `kcidb_cache_redirect()` function performs a redirection to the location of the file within the Google Cloud Storage Bucket and serve it to the user.
+
+4. **Fallback Mechanism**: In cases where the requested file is not present within the ccache storage, then `kcidb_cache_redirect()` function redirect the user to the original URL from which the file was initially requested.
+
+
```
Comment (spbnick) [inline doc/administrator_guide.md:106]: Please remove one blank line here. It's unnecessary, and we put just one blank line between sections.

**Review by spbnick — CHANGES_REQUESTED** [review]
Looks good, thank you, Abhishek! Just a couple nitpicks, inline.

File: doc/administrator_guide.md (hunk 1 above)
Comment (octonawish-akcodes) [reply doc/administrator_guide.md]: okay

File: doc/administrator_guide.md (hunk 2 above)
Comment (octonawish-akcodes) [reply doc/administrator_guide.md]: okay

**Review by spbnick — APPROVED** [review]
LGTM! Thanks, Abhishek!

**Pull request merged**